### DISCUSSION TITLE
How to activate Tailwind Intellisense for TipTap?

### DISCUSSION THREAD

**Discussion by mmailaender** [ueberdosis/tiptap#4359] [General Questions] [answered]
I found out that my Vscode Tailwind Intellisense is not working for TipTap. For my react components, the `class` attribute is working fine together with Tailwind Intellisense.
```tsx
export const TiptapExtensions = [
  StarterKit.configure({
    bulletList: {
      HTMLAttributes: {
        class: "list-disc list-outside leading-3 -mt-2", // Tailwind Intellisense not working
      },
    },
})]
```

Do I need to configure something special to get it working for TipTap too?

**Comment by amirhhashemi** [comment]
You have to configure Tailwindcss LSP to give you intellisense in the `class` attribute. I know two examples:
- [CVA](https://cva.style/docs/getting-started/installation#intellisense)
- [clsx](https://github.com/lukeed/clsx#tailwind-support)

Basically you have to write a RegExp that matches where you want to get intellisense in, and pass it to the Tailwind's LSP config of your editor..

Unfortunately I'm not good at RegExp so can't help you with that :)

**Reply by mmailaender** [reply, marked as answer]
That was a fantastic hint @amirhhashemi! This regex solves the problem:

`.vscode/settings.json`
```json
"tailwindCSS.experimental.classRegex": [
  "class:\\s*?[\"'`]([^\"'`]*).*?,"
]
```
